### PR TITLE
Final task

### PR DESCRIPTION
**Pull request by ferencbarta**

### CODE REVIEW

File: final/ticket-service/src/main/java/com/epam/training/ticketservice/Application.java [hunk 1]
```diff
@@ -0,0 +1,10 @@
+package com.epam.training.ticketservice;
+
+/**
```
Comment (BFeri92) [inline final/ticket-service/src/main/java/com/epam/training/ticketservice/Application.java:3]: Oh, I was playing with Checkstyle and forgot to remove this file afterwards.

File: final/ticket-service/pom.xml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+                            </rules>
+                        </configuration>
+                    </execution>
+                </executions>
+            </plugin>
+        </plugins>
+    </build>
+
+    <!--
+    TODO: Add checkstyle validation using Google's ruleset. Do not fail the build in case of errors.
```
Comment (ferencbarta) [inline final/ticket-service/pom.xml:136]: These are already configured, this comment should be removed.

File: final/ticket-service/pom.xml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-checkstyle-plugin</artifactId>
+                <version>${checkstyle-maven-plugin.version}</version>
+                <configuration>
+                    <configLocation>${checkstyle-config-dir}/checkstyle.xml</configLocation>
+                    <suppressionsLocation>${checkstyle-config-dir}/suppressions.xml</suppressionsLocation>
+                    <encoding>UTF-8</encoding>
+                    <consoleOutput>true</consoleOutput>
+                    <failsOnError>${requirements.fail-build-on-checkstyle-error}</failsOnError>
```
Comment (ferencbarta) [inline final/ticket-service/pom.xml:73]: `failOnViolation` should be used instead.

File: final/ticket-service-acceptance-tests/src/test/resources/features/movies.feature [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    When the user types the "update movie Sátántangó drama 450" command
+    And the user types the "list movies" command
+    Then the next line of the output is "Sátántangó (drama, 450 minutes)"
+
+  @grade2-requirement
+  Scenario: an admin user can delete a movie
+    Given the user types the "sign in privileged admin admin" command
+    And the user types the "create movie Sátántangó drama 450" command
+    When the user types the "delete movie Sátántangó" command
+    Then the next line of the output is "There are no movies at the moment"
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/movies.feature:26]: "And the user types the "list movies" command" is missing

File: final/ticket-service-acceptance-tests/src/test/resources/features/rooms.feature [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+    When the user types the "update room Pedersoli 10 10" command
+    And the user types the "list rooms" command
+    Then the next line of the output is "Room Pedersoli with 100 seats, 10 rows and 10 columns"
+
+  @grade2-requirement
+  Scenario: an admin user can delete a room
+    Given the user types the "sign in privileged admin admin" command
+    And the user types the "create room Pedersoli 20 10" command
+    When the user types the "delete room Pedersoli" command
+    Then the next line of the output is "There are no rooms at the moment"
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/rooms.feature:27]: 'And the user types the "list rooms" command' is missing

File: final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    And the user types the "create movie \"Spirited Away\" aminmation 125" command
+    And the user types the "sign out" command
+
+  @grade2-requirement
+  Scenario: The admin user can create and query screenings
+    Given the user types the "sign in privileged admin admin" command
+    When the user types the "create screening Sátántangó Pedersoli \"2021-03-15 10:45\"" command
+    And the user types the "create screening \"Spirited Away\" Pedersoli \"2021-03-14 16:00\"" command
+    And the user types the "list screenings" command
+    Then the next line of the output is "Spirited Away (animation, 125 minutes), screened in room Pedersoli, at 2021-03-14 16:00"
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature:18]: The order may be incorrect here.

File: final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature [hunk 7]
```diff
@@ -0,0 +1,57 @@
+Feature: allows creating pricing components and assigning them to rooms, movies and screenings
+
+  Background:
+    Given the application is started
+    And the prompt containing "Ticket service>" is printed
+    And the user types the "sign in privileged admin admin" command
+    And the user types the "create room Pedersoli 20 10" command
+    And the user types the "create room Girotti 10 10" command
+    And the user types the "create movie Sátántangó drama 450" command
+    And the user types the "create movie \"Spirited Away\" aminmation 125" command
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature:10]: Typo: aminamation

File: final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature [hunk 8]
```diff
@@ -0,0 +1,64 @@
+Feature: allows creating, deleting and listing screenings
+  Background: 
+    Given the application is started
+    And the prompt containing "Ticket service>" is printed
+    And the user types the "sign in privileged admin admin" command
+    And the user types the "create room Pedersoli 20 10" command
+    And the user types the "create room Girotti 10 10" command
+    And the user types the "create movie Sátántangó drama 450" command
+    And the user types the "create movie \"Spirited Away\" aminmation 125" command
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature:9]: Typo: aminmation

File: final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+    When the user types the "create screening Sátántangó Pedersoli \"2021-03-15 10:45\"" command
+    And the user types the "create screening \"Spirited Away\" Pedersoli \"2021-03-14 16:00\"" command
+    And the user types the "list screenings" command
+    Then the next line of the output is "Spirited Away (animation, 125 minutes), screened in room Pedersoli, at 2021-03-14 16:00"
+    And the next line of the output is "Sátántangó (drama, 450 minutes), screened in room Pedersoli, at 2021-03-15 10:45"
+
+  @grade2-requirement
+  Scenario: The admin can not create overlapping screenings in the same room
+    Given the user types the "sign in privileged admin admin" command
+    And the user types the "create screening Sátántangó Pedersoli 2021-03-15 10:45" command
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature:24]: Missing \"\"

File: final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    Then the next line of the output is "There is an overlapping screening"
+    When the user types the "list screenings" command
+    Then the next line of the output is "Sátántangó (drama, 450 minutes), screened in room Pedersoli, at 2021-03-15 10:45"
+
+  @grade2-requirement
+  Scenario: The admin can create overlapping screenings in different rooms
+    Given the user types the "sign in privileged admin admin" command
+    And the user types the "create screening Sátántangó Pedersoli 2021-03-15 10:45" command
+    When the user types the "create screening \"Spirited Away\" Girotti \"2021-03-15 10:50\"" command
+    Then the next line of the output is "There is an overlapping screening"
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature:35]: There should not be an error message in this case

File: final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+    And the next line of the output is "Spirited Away (animation, 125 minutes), screened in room Girotti, at 2021-03-15 10:50"
+
+  @grade2-requirement
+  Scenario: The admin can not create screenings during a 10 minute long breaks after another screening in the same room
+    Given the user types the "sign in privileged admin admin" command
+    And the user types the "create screening Sátántangó Pedersoli 2021-03-15 11:00" command
+    When the user types the "create screening \"Spirited Away\" Pedersoli \"2021-03-15 18:39\"" command
+    Then the next line of the output is "This would start in the break period after another screening in this room"
+    When the user types the "list screenings" command
+    Then the next line of the output is "Sátántangó (drama, 450 minutes), screened in room Pedersoli, at 2021-03-15 10:45"
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/screenings.feature:47]: This actually ends on 2021-03-15 11:00

File: final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature [hunk 12]
```diff
@@ -0,0 +1,57 @@
...
+    And the user types the "create movie \"Pulp Fiction\" drama 154" command
+    And the user types the "create screening Sátántangó Pedersoli \"2021-03-15 10:45\"" command
+    And the user types the "create screening \"Spirited Away\" Pedersoli \"2021-03-14 16:00\"" command
+    And the user types the "create screening \"Pulp Fiction\" Girotti \"2021-03-14 16:00\"" command
+    And the user types the "sign out" command
+
+  @grade2-requirement
+  Scenario: a default per seat per screening price is defined to be 1500 HUF and is used in the calculation of the price of each seat
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature:18]: This is required for booking and hence should be a @grade4-requirement

File: final/ticket-service-acceptance-tests/src/test/resources/features/bookings.feature [hunk 13]
```diff
@@ -0,0 +1,51 @@
...
+    And the user types the "create movie Sátántangó drama 450" command
+    And the user types the "create movie \"Spirited Away\" animation 125" command
+    And the user types the "create screening \"Spirited Away\" Girotti \"2021-03-15 10:00\"" command
+    And the user types the "create screening Sátántangó Pedersoli \"2021-03-15 10:45\"" command
+    And the user types the "sign out" command
+
+  @grade4-requirement
+  Scenario: an authenticated user can book a ticket for a screening
+    Given the user types the "sign in sanyi asdQWE123" command
+    When the user types the "book Sátántangó Pedersoli \"2021-03-15 10:45\" (5,5) (5,6)" command
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/bookings.feature:20]: Having an unlimited number of parameters looks quite problematic; we should wrap (5,5) (5,6) in \" \".

File: final/README.md [hunk 14]
```diff
@@ -0,0 +1,447 @@
...
+    ```
+* A parancs kimenete más esetekben a kettes jegyhez szükséges követelmények között van leírva.
+
+##### Jegyfoglalás
+
+###### Foglalás létrehozása
+* A következő parancs lehetőséget biztosít egy már meglévő vetítésre, egy vagy több ülőhelyre
+jegyet foglalni.
+```
+book <film címe> <terem neve> <vetítés kezdetének ideje YYYY-MM-DD hh:mm formátumban> <a lefoglalandó ülőhelyek listája, szóközzel elválasztva, egy-egy ülőhely "(<sor>,<oszlop>)" formátumú>
```
Comment (ferencbarta) [inline final/README.md:332]: I'm afraid using parentheses won't work here - jline seem to have a widget for closing parenthesis which, in our case, prints a part of the input.

Comment (ferencbarta) [reply final/README.md:332]: Also, the list of seats should be wrapped in ""

File: final/README.md [hunk 15]
```diff
@@ -0,0 +1,447 @@
...
+a `ticket-service-parent` projekten. Ehhez a négyes érdemjegy eléréséhez szükséges 
+feltételeken túl a következők szükségesek:
+    * Legalább 70%-os unit teszt branch és line coverage az egész `ticket-service` modulra vonatkoztatva.
+    * A `@grade5-requirements` tag-gel megjelölt acceptance tesztekben leírt követelményeknek megfelelő alkalmazás.
+
+#### Az alkalmazás által támogatott parancsok
+
+##### Árazással kapcsolatos funkcionalitás
+
+###### Az alapértelmezett ár megváltoztatása
```
Comment (ferencbarta) [inline final/README.md:371]: Calling it "Alapár" / base price would be better.

File: final/README.md [hunk 16]
```diff
@@ -0,0 +1,447 @@
...
+* Egy foglalás alapára 1500 HUF / ülőhely / vetítés. 
+A foglalás ára (/ülőhely/vetítés) kiszámítható a foglalás alapárának 
+és a foglalásra vonatkozó árkomponensek összegzésével. Az árkomponensek
+leírását lásd később. 
+* A következő parancs a foglalás alapárának megváltoztatását teszi lehetővé
+```
+update default price <új alapértelmezett ár> HUF
```
Comment (ferencbarta) [inline final/README.md:378]: I guess we can remove the currency; the requirement does not mention handling foreign currency

File: final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature [hunk 17]
```diff
@@ -0,0 +1,57 @@
...
+    And the user types the "create room Girotti 10 10" command
+    And the user types the "create movie Sátántangó drama 450" command
+    And the user types the "create movie \"Spirited Away\" aminmation 125" command
+    And the user types the "create movie \"Pulp Fiction\" drama 154" command
+    And the user types the "create screening Sátántangó Pedersoli \"2021-03-15 10:45\"" command
+    And the user types the "create screening \"Spirited Away\" Pedersoli \"2021-03-14 16:00\"" command
+    And the user types the "create screening \"Pulp Fiction\" Girotti \"2021-03-14 16:00\"" command
+    And the user types the "sign out" command
+
+  @grade2-requirement
```
Comment (ferencbarta) [inline final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature:17]: Querying the price is a grade 5 requirement

File: final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature (hunk 12 above)
Comment (ferencbarta) [reply final/ticket-service-acceptance-tests/src/test/resources/features/pricing.feature:18]: Actually, this requires show-price-for, which is grade5.

**Review by lamfalusy — APPROVED** [review]

**Pull request merged**